service/controller/release/resource_set.go

```diff
@@ -0,0 +1,129 @@
+package release
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/pkg/clientset/versioned"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/controller"
+	"github.com/giantswarm/operatorkit/controller/resource/metricsresource"
+	"github.com/giantswarm/operatorkit/controller/resource/retryresource"
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/giantswarm/release-operator/service/controller/release/controllercontext"
+	"github.com/giantswarm/release-operator/service/controller/release/resource/app"
+	"github.com/giantswarm/release-operator/service/controller/release/resource/status"
+)
+
+type ResourceSetConfig struct {
+	G8sClient versioned.Interface
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+
+	ProjectName string
+}
+
+func NewResourceSet(config ResourceSetConfig) (*controller.ResourceSet, error) {
+	var err error
+
+	var statusResource controller.Resource
+	{
+		c := status.Config{
+			G8sClient: config.G8sClient,
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			Namespace: "giantswarm",
+		}
+
+		statusResource, err = status.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var appResource controller.Resource
+	{
+		c := app.Config{
+			G8sClient: config.G8sClient,
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+
+			Namespace: "giantswarm",
+		}
+
+		appResource, err = app.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []controller.Resource{
+		statusResource,
+		appResource,
```

Comment:
Can you elaborate on the app resource? I don't follow what this is doing.

Reply:
Yes. `app` resource in `release` controller will create App CR for each component in the release. E.g. it will install `aws-operator.v1.2.3`.

`app` resource in `releasecycle` controller will create App CR for release which observed ReleaseCycle CR describes. E.g. `aws.v2.3.4`.

Reply:
Late to the party, but about naming: how about having
- `appComponentResource` which creates components App CR
- `appReleaseResource` which creates release App CR

wdyt?

Reply:
They will be in separate controllers so we can implement those two things in separate resources without having a name clash.